client/src/app/containers/Game/saga.js

```diff
@@ -0,0 +1,174 @@
+  yield call(history.push, '/home');
+}
+
+export function* handleGoToGame({ payload: { id, history } }) {
+  if (!id) {
+    yield put(
+      alertActions.addAlert('No id in handleGoToGame', alertTypes.error),
+    );
+    return;
+  }
+
+  const currentUser = yield select(selectCurrentUser);
+
+  // select the id being used to login from the state
+  try {
+    const res = yield axios.get(`/game/${id}?player=${currentUser.id}`);
+    yield put(gameActions.gameRetrieved({ data: res.data }));
+    yield call(history.push, '/game');
+  } catch (err) {
+    yield put(
+      alertActions.addAlert(
+        `something bad happened... ${err}`,
+        alertTypes.error,
+      ),
+    );
+  }
+}
+
+export function* handleRefreshCurrentGame({ payload: { id, history } }) {
+  const currentUser = yield select(selectCurrentUser);
+
+  try {
+    const res = yield axios.get(`/game/${id}?player=${currentUser.id}`);
+    yield put(gameActions.gameRetrieved({ data: res.data }));
+  } catch (err) {
+    yield put(alertActions.addAlert(err.response.data, alertTypes.error));
+  }
+}
+
+const cardToGolangCard = c => {
+  const magicMap = {
+    Spades: 0,
+    Clubs: 1,
+    Diamonds: 2,
+    Hearts: 3,
+  };
+  return {
+    s: magicMap[c.suit],
+    v: c.value,
+  };
+};
+
+const getPlayerActionJSON = (myID, gID, phase, currentAction) => {
+  let overcomesMap = {
+    deal: 0,
+    crib: 1,
+    cut: 2,
+    peg: 3,
+    counthand: 4,
+    countcrib: 5,
+  };
+  let action = {};
+  switch (phase) {
+    case 'deal':
+      action = { ns: currentAction.numShuffles };
+      break;
+    case 'crib':
+      action = {
+        cs: currentAction.selectedCards
+          ? currentAction.selectedCards.map(cardToGolangCard)
+          : [],
+      };
+      break;
+    case 'cut':
+      action = { p: currentAction.percCut };
+      break;
+    case 'peg':
+      action =
+        !currentAction.selectedCards || currentAction.selectedCards.length !== 1
+          ? {
+              sg: true,
+            }
+          : {
+              c: cardToGolangCard(currentAction.selectedCards[0]),
+            };
+      break;
+    case 'counthand':
+      action = { pts: currentAction.points };
+      break;
+    case 'countcrib':
+      action = { pts: currentAction.points };
+      break;
+    default:
+      action = { badstate: true };
+      break;
+  }
+  return {
+    pID: myID,
+    gID: gID,
+    o: overcomesMap[phase],
+    a: action,
+  };
+};
+
+function* handleGenericAction(phase) {
+  const currentUser = yield select(selectCurrentUser);
+  const id = yield select(selectCurrentGameID);
+  const currentAction = yield select(selectCurrentAction);
+
+  try {
+    // the return of the post is just 'action handled'
+    // it may be wise to make it return the new game state?
+    yield axios.post(
+      '/action',
+      getPlayerActionJSON(currentUser.id, id, phase, currentAction),
+    );
+    // TODO wait a moment and re-fetch?
```

Comment:
Idk about that. It's going to be 50/50 whether or not it catches the opponent's action. Can we just wait until we have actual websockets implemented?

Reply:
I was/am concerned that the server will get the "handle action request", start handling it, and then get the "get game state" request, start handling that before the action has been completed. But it's seeming to work for me, so :yolo: